infer_segmentation.py

```diff
@@ -139,7 +141,19 @@ def segment_image(model, image, prediction_threshold, target_size_1d, background
 
 
 def main(gcp_bucket, model_id, background_class_index, stack_id, image_ids, user_specified_prediction_thresholds,
-         labels_output, pad_output, trained_thresholds_id):
+         labels_output, pad_output, trained_thresholds_id, python_random_global_seed, numpy_random_global_seed,
+         tf_random_global_seed):
+
+    # seed global random generators if specified; global random seeds here must be convertible to int or exactly 'None'
```

Comment:
argparser limitations. cannot have type `int` and defulat to an `int`, but also allow `None` input. `str` allows it all and checks are made later

Reply:
ah right. got it. forgot about this drama. my suggestion then would be to use an empty string to signify "no seed given" - sorry if this is feeling nitpicky, this is just the sort of thing imo that can created a bunch of confusion

Reply:
@Josh-Joseph so to not seed and thus be random (given that default is a seed), user should input `--random-module-global-seed ''`?

Reply:
I think they'd just not include that flag in the command line and the default value in argparse would be `''`

Reply:
@Josh-Joseph i think default should be seeded, correct?

Reply:
oh interesting! I was assuming no. That's not typically the norm but if you think that's how you'd like to set it up it's good w/ me. If you do that I'd at least include a warning like `print('Warning: since no random seed was given, using the default random seed')` since imo most people wouldn't expect that behavior

Reply:
@Josh-Joseph ooohhhhhh gotcha, defaulting to a seed is what's created all of the confusion here. much easier to default to no seed. If no seed is the norm, let's stick with that. then the argparser will simply be changed from str to int and default to `None`, sound good?

Reply:
cool cool sounds good to me

Reply:
@Josh-Joseph okay done. much much easier this way, wish i would've clarified this with you at the onset :-D

Reply:
all good - wish I would've been more clear too about what I was assuming but glad we got to the bottom of it!